### PR TITLE
2.charts updated

### PR DESCRIPTION
**Pull request by AKHIL-882**

### CODE REVIEW

File: CoinDock-api/routes/v1/users.php [hunk 1]
```diff
@@ -51,15 +57,19 @@
...
+                Route::get('/piechartdata', [WalletCoinController::class, 'showPieChartData'])->missing(
                     fn () => response(
                         [
                             'error' => ['message' => 'User record not found'],
                         ],
                         404,
                     ),
                 );
-
+                // Route::get('/real-time-graph/display/', [WalletCoinController::class, 'displaySingleCoinHistoricalData']);
```
Comment (bharatjarvis) [inline CoinDock-api/routes/v1/users.php]: please remove uncommented code if not needed

File: CoinDock-api/database/migrations/2022_05_17_161749_create_coins_table.php [hunk 2]
```diff
@@ -15,7 +15,11 @@ public function up()
     {
         Schema::create('coins', function (Blueprint $table) {
             $table->id();
+            $table->string('coin_id');
```
Comment (bharatjarvis) [inline CoinDock-api/database/migrations/2022_05_17_161749_create_coins_table.php]: Don't update existing migrations. If any new changes were required then introduce a separate migration for the  changes.

File: CoinDock-api/routes/v1/users.php (hunk 1 above)
Comment (AKHIL-882) [reply CoinDock-api/routes/v1/users.php]: Okay. I will update the code.

File: CoinDock-api/database/migrations/2022_05_17_161749_create_coins_table.php (hunk 2 above)
Comment (AKHIL-882) [reply CoinDock-api/database/migrations/2022_05_17_161749_create_coins_table.php]: Okay.

**Comment by bharatjarvis** [comment]
Please make sure your is in sync with main and resolve any conflicts

File: CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Console\Commands\historical_data;
+
```
Comment (sita7-debug) [inline CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php]: it's HistoricalData not historical_data

File: CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+use App\Models\V1\User;
+use Carbon\Carbon;
+use Illuminate\Console\Command;
+
+class handleHistoricalData extends Command
+{
```
Comment (sita7-debug) [inline CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php]: here HandleHistorticaData not handleHistorticaData

File: CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    public function handle()
+    {
+        // $user = new User();
+        // $end_date = Carbon::now()->format('Y-m-d');
+        // $end_date = $end_date."T".Carbon::now()->format('H:i:m');
+        // $range = '7DAY';
+        // $start_date = Carbon::now()->subYear(1);
+        // $start_date  = str_replace(' ','T',$start_date);
+        // $realTimeData = $user->commonDataRealTime('BTC',$range,$start_date,$end_date);
+        // $assetArray = collect(json_decode($realTimeData));
```
Comment (sita7-debug) [inline CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php]: Is this command do any specific job? (But here everything was commented) if not then please remove the unnecessary code file.

File: CoinDock-api/app/Http/Controllers/V1/Auth/AuthController.php [hunk 6]
```diff
@@ -33,6 +33,7 @@ public function login(LoginRequest $request)
             [
                 'message' => 'Login Successfull.',
                 'results' => [
+                    'token'=>$response['access_token'],
                     'user' => UserResource::make($user)->resolve(),
```
Comment (sita7-debug) [inline CoinDock-api/app/Http/Controllers/V1/Auth/AuthController.php]: token should be in handle under headers section, not in results section.

File: CoinDock-api/app/Models/V1/User.php [hunk 7]
```diff
@@ -91,6 +95,352 @@ public function store(CreateUserRequest $request): self
...
+                    }
+                }
+            }
+            foreach($wallets as $key1=>$value1){
+                foreach($coinShortName as $shortName){
+                    $primaryCurrency = Setting::whereUserId($user->id)->first()->primary_currency;
+                    $baseURL = config('cryptohistoricaldata.coin.api_url').config('cryptohistoricaldata.coin.exchange_url');
+                    $baseURLId1Replace = str_replace('{id1}', $shortName, $baseURL);
+                    $baseURLId1Replaced = $baseURLId1Replace;
+                    $baseURLIdReplaced = str_replace('{id2}', $primaryCurrency, $baseURLId1Replaced);
```
Comment (sita7-debug) [inline CoinDock-api/app/Models/V1/User.php]: str_replace allows arrays. please refer: [https://www.php.net/manual/en/function.str-replace.php](url)

File: CoinDock-api/app/Models/V1/User.php [hunk 8]
```diff
@@ -91,6 +95,352 @@ public function store(CreateUserRequest $request): self
...
+            foreach($wallets as $key1=>$value1){
+                foreach($coinDataAll as $coinData){
+                    if(strtolower($coinData->name) == strtolower($key1)){
+                        array_push($coinShortName,$coinData->coin_id);
+                    }
+                }
+            }
+            foreach($wallets as $key1=>$value1){
+                foreach($coinShortName as $shortName){
+                    $primaryCurrency = Setting::whereUserId($user->id)->first()->primary_currency;
```
Comment (sita7-debug) [inline CoinDock-api/app/Models/V1/User.php]: use relations instead of using DB call

File: CoinDock-api/app/Models/V1/User.php [hunk 9]
```diff
@@ -91,6 +95,352 @@ public function store(CreateUserRequest $request): self
 
     }
 
+    public function showPieChartData(User $user, Request $request){
+        $filterByCondition = $request->filterBy;
+        $wallets = Wallet::select(['coin_id', 'balance'])
+            ->whereUserId($user->id)
```
Comment (sita7-debug) [inline CoinDock-api/app/Models/V1/User.php]: use relation instead of DB call

File: CoinDock-api/app/Models/V1/User.php [hunk 10]
```diff
@@ -91,6 +95,352 @@ public function store(CreateUserRequest $request): self
...
+                'message'=>'success',
+                'data'=>$result
+            ],200);
+            
+        }
+    }
+
+    public function showUserCoins(User $user){
+        $coinNames = Wallet::whereUserId($user->id)->get();
+        $coinIds = [];
```
Comment (sita7-debug) [inline CoinDock-api/app/Models/V1/User.php]: same here

File: CoinDock-api/app/Models/V1/User.php [hunk 11]
```diff
@@ -91,6 +95,352 @@ public function store(CreateUserRequest $request): self
...
+        foreach ($coinNames as $coinName){
+            array_push($coinIds,$coinName->coin_id);
+        }
+        $coinIds =array_unique($coinIds);
+        $coinList=[];
+        foreach ($coinIds as $coinId){
+            $coinName = Coin::whereId($coinId)->first();
```
Comment (sita7-debug) [inline CoinDock-api/app/Models/V1/User.php]: same here

File: CoinDock-api/app/Models/V1/User.php [hunk 12]
```diff
@@ -91,6 +95,352 @@ public function store(CreateUserRequest $request): self
...
+        $range = $request->range;
+        $end_date = Carbon::now()->format('Y-m-d');
+        $end_date = $end_date."T".Carbon::now()->format('H:i:m');
+        if($coinNameFromUser==NULL){
+            if($range==NULL){
+                $coinIdList = $this->returnCoinId($user);
+                return $this->dayDataForAllCoins($coinIdList);
+            }else{
+                if($range=='DAY'){
+                    $coinIdList = $this->returnCoinId($user);
```
Comment (sita7-debug) [inline CoinDock-api/app/Models/V1/User.php]: instead of using if-elseif use switch

File: CoinDock-api/app/Models/V1/User.php [hunk 13]
```diff
@@ -100,4 +450,8 @@ public function signUp()
     {
         return $this->hasOne(Signup::class);
     }
+
+    public function wallet(){
```
Comment (sita7-debug) [inline CoinDock-api/app/Models/V1/User.php]: if the relation was hasMany() then the function name should be plural.

**Review by sita7-debug — CHANGES_REQUESTED** [review]

File: CoinDock-api/app/Models/V1/User.php (hunk 7 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Models/V1/User.php]: Modified

File: CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php (hunk 3 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php]: Handler Removed

File: CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php (hunk 4 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php]: Handler Removed

File: CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php (hunk 5 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Console/Commands/historical_data/handleHistoricalData.php]: Handler Removed

File: CoinDock-api/app/Http/Controllers/V1/Auth/AuthController.php (hunk 6 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Http/Controllers/V1/Auth/AuthController.php]: Modified

File: CoinDock-api/app/Models/V1/User.php (hunk 12 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Models/V1/User.php]: Modified

File: CoinDock-api/app/Models/V1/User.php (hunk 13 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Models/V1/User.php]: Modified

File: CoinDock-api/app/Models/V1/User.php (hunk 10 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Models/V1/User.php]: Modified

**Review by sita7-debug — APPROVED** [review]

**Review by bharatjarvis — APPROVED** [review]

**Pull request merged**